### PR TITLE
fix: Collapse Panel key type definition

### PR DESCRIPTION
**Pull request by thylsky**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?
1. Describe the source of requirement, like related issue link.
    Currently the allowed key for the `Collapse.Panel` is just string.

2. Describe the problem and the scenario.
    When you are rendering `Collapse.Panel` with map function, you can't simply provide just index of the map function. Also React's definition for Key type is `number | string`. 

### 💡 Solution

1. How to fix the problem, and list final API implementation and usage sample if that is an new feature.
    Fix is prepared in this PR: Changing type definition of the `key` props.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |          |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/collapse/index.en-US.md](https://github.com/thylsky/ant-design/blob/patch-1/components/collapse/index.en-US.md)
[View rendered components/collapse/index.zh-CN.md](https://github.com/thylsky/ant-design/blob/patch-1/components/collapse/index.zh-CN.md)

### CODE REVIEW

**Comment by dengfuping** [comment]
@thylsky Remember to rebase it.

**Comment by thylsky** [comment]
> @thylsky Remember to rebase it.

Fixed. 👍

**Comment by dengfuping** [comment]
@thylsky `activeKey` and `defaultActiveKey` should be same with `key`.
image

**Comment by thylsky** [comment]
> @thylsky `activeKey` and `defaultActiveKey` should be same with `key`.
> image

Good catch, thanks! Resolved.

File: components/collapse/Collapse.tsx [hunk 1]
```diff
@@ -9,8 +9,8 @@ import animation from '../_util/openAnimation';
 export type ExpandIconPosition = 'left' | 'right';
 
 export interface CollapseProps {
-  activeKey?: Array<string> | string;
-  defaultActiveKey?: Array<string>;
+  activeKey?: Array<string> | string | number;
+  defaultActiveKey?: Array<string | number>;
```
Comment (dengfuping) [inline components/collapse/Collapse.tsx]: @thylsky Should be:
```
activeKey?: Array<string | number> | string | number;
defaultActiveKey?: Array<string | number> | string | number;
```

Comment (thylsky) [reply components/collapse/Collapse.tsx]: Yes, it was kinda confusing to me, why the defaultActiveKey expects only array.

Comment (thylsky) [reply components/collapse/Collapse.tsx]: But the documentation says that it doesn't expects an array...

Comment (dengfuping) [reply components/collapse/Collapse.tsx]: @thylsky activeKey supports single value and array value both and `defaultActiveKey`'s type is as same as `activeKey`.

Comment (thylsky) [reply components/collapse/Collapse.tsx]: @dengfuping Fixed. 👍I've also updated the documentation related to this.

File: components/collapse/CollapsePanel.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import classNames from 'classnames';
 import { ConfigConsumer, ConfigConsumerProps } from '../config-provider';
 
 export interface CollapsePanelProps {
-  key: string;
+  key: string | number;
```
[the remark this answers is not in the document]
Comment (thylsky) [reply components/collapse/CollapsePanel.tsx]: I don't think it's necessary to add the activeKey and the defaultActiveKey here. It should be defined in the Collapse parent component.

**Review by dengfuping — APPROVED** [review]

**Pull request merged**